clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -688,6 +722,7 @@ public void abortTransaction() throws ProducerFencedException {
      * @param callback A user-supplied callback to execute when the record has been acknowledged by the server (null
      *        indicates no callback)
      *
+     * @throws IllegalStateException if a transactional.id has been configured and no transaction has been started
```

Comment:
Oh you mean `UnsupportedForMessageFormatException`? That doesn't seem to be added.